tests/rptest/tests/redpanda_oauth_test.py

```diff
@@ -184,6 +187,105 @@ def check_sr_subjects():
 
         wait_until(check_sr_subjects, timeout_sec=5)
 
+    @cluster(num_nodes=4)
+    def test_java_client(self):
+        kc_node = self.keycloak.nodes[0]
+
+        self.keycloak.admin.create_user('norma',
+                                        'desmond',
+                                        realm_admin=True,
+                                        email='10086@sunset.blvd')
+        self.keycloak.login_admin_user(kc_node, 'norma', 'desmond')
+        self.keycloak.admin.create_client(CLIENT_ID)
+
+        # add an email address to myapp client's service user. this should
+        # appear alongside the access token.
+        self.keycloak.admin.update_user(f'service-account-{CLIENT_ID}',
+                                        email='myapp@customer.com')
+
+        self.rpk.create_topic(EXAMPLE_TOPIC)
+        service_user_id = self.keycloak.admin_ll.get_user_id(
+            f'service-account-{CLIENT_ID}')
+        result = self.rpk.sasl_allow_principal(f'User:{service_user_id}',
+                                               ['all'], 'topic', EXAMPLE_TOPIC,
+                                               self.su_username,
+                                               self.su_password,
+                                               self.su_algorithm)
+
+        cfg = self.keycloak.generate_oauth_config(kc_node, CLIENT_ID)
+        assert cfg.client_secret is not None
+        assert cfg.token_endpoint is not None
+
+        cli = KafkaCliTools(self.redpanda,
+                            user=self.su_username,
+                            passwd=self.su_password,
+                            oauth_cfg=cfg)
+        expected_topics = set([EXAMPLE_TOPIC])
+        wait_until(lambda: set(cli.list_topics()) == expected_topics,
+                   timeout_sec=5)
+
+        N_REC = 10
+        cli.oauth_produce(EXAMPLE_TOPIC, N_REC)
+        records = []
+        for i in range(0, N_REC):
+            rec = self.rpk.consume(EXAMPLE_TOPIC, n=1, offset=i)
+            records.append(json.loads(rec))
+
+        self.redpanda.logger.debug(json.dumps(records, indent=1))
+
+        assert len(
+            records) == N_REC, f"Expected {N_REC} records, got {len(records)}"
+
+        values = set([r['value'] for r in records])
+
+        assert len(values) == len(
+            records
+        ), f"Expected {len(records)} unique records, got {len(values)}"
+
+    @cluster(num_nodes=4)
+    def test_java_client_unauthorized(self):
```

Comment:
This duplicates a lot of the test above. Do you think it would be cleaner to attempt a produce or consume with the oidc credentials before the ACL is added and ensure an auth failure is returned?

Or perhaps parametrize / matrix on the test?

Reply:
yeah fair point